xDrip/Scenes/Entries/EntriesList/EntriesListViewController.swift

```diff
@@ -59,9 +74,89 @@ class EntriesListViewController: UIViewController, EntriesListDisplayLogic {
         interactor?.doLoad(request: request)
     }
     
+    @IBAction private func onCancelButtonTap() {
+        let request = EntriesList.Cancel.Request()
+        interactor?.doCancel(request: request)
+    }
+    
+    @IBAction func onEditButtonTap() {
+        isEdit.toggle()
+    }
+    
+    
     // MARK: Display
     
     func displayLoad(viewModel: EntriesList.Load.ViewModel) {
+        cellData = viewModel.cellData
+        tableView.reloadData()
+    }
+    
+    private func setupUI() {
+        editBarButtonItem.title = "edit".localized
+        
+        view.addBlur()
+    }
+}
+
+// MARK: - UITableViewDataSource
+
+extension EntriesListViewController: UITableViewDataSource {
```

Comment:
Usually containing table view delegate and datasource in view controller is ok, but in this case there's too much logic for it. Move it to a separate class `EntriesListTableController`. Move editing logic there as well. Use closure to delegate events back to view controller.